client/authority-discovery/src/worker.rs

```diff
@@ -232,6 +233,25 @@ where
 		}
 	}
 
+	fn addresses_to_publish(&self) -> impl ExactSizeIterator<Item = Multiaddr> {
```

Comment:
> We need an extra dependency for it

Within Substrate we already depend on `either` 13 times, in addition, without the `derive` feature `either` doesn't have any dependencies. Is this not a trade off worth taking?

> It will be collected anyway, which means in the case with sentry nodes, we clone the structure, convert it into an iterator and create a vector of it afterwards again. This could be solved with a simple clone.

True. In the case of sentry nodes this is sub optimal, but in the case of a validator itself it removes the need for one Iterator conversion (i.e. `Vec -> Iterator -> filter -> inner to_vec ->  Vec` instead of `Vec -> Iterator -> filter -> Vec -> Iterator -> inner to vec -> Vec`). With the fact in mind that sentry nodes are deprecated (https://github.com/paritytech/substrate/issues/6845) I think optimizing for the latter makes more sense. Lastly I very much doubt this has an impact in the first place as it is executed every 12 h only.

Does the above make sense @bkchr?

Reply:
Yeah fine by me.

This was also no blocker by me ;)